codalab/codalabtools/compute/worker.py

```diff
@@ -297,6 +298,16 @@ def run(task_id, task_args):
         detailed_results_url = task_args['detailed_results_url']
         private_output_url = task_args['private_output_url']
 
+        """
+        DOCKER IMAGE SANITATION
+        """
+
+        # Remove all excess whitespaces on edges, split on spaces and grab the first word.
+        sanitized_docker_image = '"{}"'.format(docker_image.strip().split(' ')[0])
+        # Regex acts as a whitelist here. Only alphanumerics and the following symbols are allowed: / . : -.
+        # If any not allowed are found, replaced with second argument to sub.
+        sanitized_docker_image = re.sub('[^0-9a-zA-Z/.:-]+', '', sanitized_docker_image)
```

Comment:
As a fun practice exercise, let's add some tests for this -- we can pair program it tomorrow!